core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java

```diff
@@ -0,0 +1,107 @@
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+
+@ForestApplication
+class PreHandlerTestApp {
+
+    @PreHandler(value = "/prehandler/*")
+    public void preHandle(HttpServerRequest request, HttpServerResponse response) {
+        response.write(request.path(), "UTF-8");
+    }
+
+    @PreHandler("/prehandler/error/*")
+    public boolean preHandleException(HttpServerRequest request, HttpServerResponse response) {
+        response.setStatusCode(503).write("Service Unavailable for " + request.path());
+        return false;
+    }
+
+    @Route(path = "/prehandler/*", type = RoutingType.HANDLER)
+    public void handle(HttpServerResponse response) {
+        response.write(" is handled", "UTF-8");
+    }
+
+    @Route(path = "/prehandler/error/*", type = RoutingType.HANDLER)
+    public void handleException(HttpServerResponse response) {
+        response.write(" should not be here", "UTF-8");
+    }
+}
+
+@DisplayName("When pre-handler methods default")
+@ExtendWith(ForestExtension.class)
+@ForestTest(appClass = PreHandlerTestApp.class)
+@DisableAutoScan
+public class PreHandlerIntegrationTest {
+
+    @Inject
+    @Config("forest.http.port")
+    Integer port;
+
+    CloseableHttpClient client;
+
+    @BeforeEach
+    void setUp() {
+        client = HttpClients.createDefault();
+    }
+
+    @ParameterizedTest(name = "should handle \"{0}\" method")
+    @CsvSource({
+            "GET, /prehandler/get, /prehandler/get is handled",
+            "POST, /prehandler/post, /prehandler/post is handled",
+            "PATCH, /prehandler/patch, /prehandler/patch is handled",
+            "DELETE, /prehandler/delete, /prehandler/delete is handled"
+    })
+    void should_handle_all_methods_when_methods_attribute_is_default_value_in_preHandler(String method, String path, String expected) throws IOException {
+        Assertions.assertEquals(expected, sendRequestandGetResult(method, path));
+    }
+
+    @ParameterizedTest(name = "should handle \"{0}\" and return")
+    @CsvSource({
+            "GET, /prehandler/error/get, Service Unavailable for /prehandler/error/get",
+            "POST, /prehandler/error/post, Service Unavailable for /prehandler/error/post",
+            "PATCH, /prehandler/error/patch, Service Unavailable for /prehandler/error/patch",
+            "DELETE, /prehandler/error/delete, Service Unavailable for /prehandler/error/delete"
+    })
+    void should_handle_all_methods_when_methods_attribute_is_default_value_in_preHandler_and_return_when_error_occurs(String method, String path, String expected) throws IOException {
+        CloseableHttpResponse response = sendRequestGetResponse(method, path);
+        int statusCode = response.getStatusLine().getStatusCode();
+
+        Assertions.assertEquals(503, statusCode);
+        Assertions.assertEquals(expected, sendRequestandGetResult(method, path));
+        Assertions.assertFalse(sendRequestandGetResult(method, path).contains("should not be here"));
```

Comment:
Why don't you request once, assert twice? What you're doing right now is wasting time.

Reply:
'kay, will do.